Makefile

```diff
@@ -52,22 +67,53 @@ endif
 # List of smoke test steps.  This is used by CI
 #
 
-smoke-test:	test-clean-up	build_test
-	$(TEST_BIN) smoke --spu ${DEFAULT_SPU} --local ${TEST_LOG} -r ${REPL} ${SKIP_CHECK} ${EXTRA_ARG} -- --producer-iteration=${DEFAULT_ITERATION}
-
-smoke-test-stream:	test-clean-up	build_test
-	$(TEST_BIN) smoke --spu ${DEFAULT_SPU} --local ${TEST_LOG} ${SKIP_CHECK} -- --consumer-wait=true --producer-iteration=${DEFAULT_ITERATION}
-
-smoke-test-tls:	test-clean-up build_test
-	$(TEST_BIN) smoke --spu ${DEFAULT_SPU} --tls --local ${TEST_LOG} ${SKIP_CHECK} -- --producer-iteration=${DEFAULT_ITERATION}
-
-smoke-test-tls-policy:	test-clean-up build_test
-	AUTH_POLICY=$(SC_AUTH_CONFIG)/policy.json X509_AUTH_SCOPES=$(SC_AUTH_CONFIG)/scopes.json  \
-	FLV_SPU_DELAY=$(SPU_DELAY) \
-	$(TEST_BIN) smoke --spu ${DEFAULT_SPU} --tls --local ${TEST_LOG} ${SKIP_CHECK} --keep-cluster -- --producer-iteration=${DEFAULT_ITERATION}
+smoke-test-base: test-setup smoke-test-base-nobuild
+smoke-test-base-nobuild: test-setup-nobuild
+	# Set ENV
+	AUTH_POLICY=$(TEST_ENV_AUTH_POLICY) \
+	FLV_SPU_DELAY=$(TEST_ENV_FLV_SPU_DELAY) \
+		$(TEST_BIN) smoke \
+			${TEST_ARG_SPU} \
+			${TEST_ARG_LOG} \
+			${TEST_ARG_TLS} \
+			${TEST_ARG_LOCAL} \
+			${TEST_ARG_DEVELOP} \
+			${TEST_ARG_REPLICATION} \
+			${TEST_ARG_KEEP_CLUSTER} \
+			${TEST_ARG_AUTH_CONFIG_MAP} \
+			${TEST_ARG_SKIP_CHECKS} \
+			${TEST_ARG_EXTRA} \
+			-- \
+			${TEST_ARG_CONSUMER_WAIT} \
+			${TEST_ARG_PRODUCER_ITERATION}
+
+smoke-test-nobuild: TEST_ARG_LOCAL=--local
+smoke-test-nobuild: TEST_ARG_SKIP_CHECKS=--skip-checks
+smoke-test-nobuild: smoke-test-base-nobuild
+smoke-test: test-setup smoke-test-nobuild
+
+smoke-test-stream-nobuild: TEST_ARG_SKIP_CHECKS+=--skip-checks
```

Comment:
ditto.  `smoke-test-stream`

Reply:
Did you see that each recipe has two variants? There is `smoke-test-stream-nobuild` and `smoke-test-stream`, which _will_ build. It's expected that developers will mostly use `smoke-test-stream` and CI will use `smoke-test-stream-nobuild`